app/src/main/java/ch/epfl/skysync/viewmodel/CalendarViewModel.kt

```diff
@@ -159,4 +159,23 @@ class CalendarViewModel(
         // However this might be unnecessary
         refreshUser()
       }
+
+  /** Cancels the modification made in the current availability calendar */
+  fun cancelAvailabilities() =
```

Comment:
couldn't we just reset the availabilities of the user with `originalAvailabilityCalendar` that contains the persitent state of the AvailabilityCalendar before the modifications?

Reply:
If you're thinking of `user.availabilities = originalAvailabilityCalendar`, this isn't possible because user.availabilities is immutable. But if you're thinking of something else, I'd like to know.

Reply:
why is that a problem?

Reply:
User.availabilities is val and can not be reassigned.

Reply:
and making just a copy of the user with the new availabilities?